# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

#try:
#    data = pd.read_csv('/datasets/data.csv')
#except:
#    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')
    
data = pd.read_csv('/Users/alinavayner/Desktop/Data_Analyst/Проекты/проект_2_ кредитный_скоринг/data.csv')
data.to_csv('data.csv', index=False) 

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()


children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)
data.head(40)

children  days_employed  dob_years            education  education_id  \
0          1    8437.673028         42               высшее             0   
1          1    4024.803754         36              среднее             1   
2          0    5623.422610         33              среднее             1   
3          3    4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0     926.185831         27               высшее             0   
6          0    2879.202052         43               высшее             0   
7          0     152.779569         50              среднее             1   
8          2    6929.865299         35               высшее             0   
9          0    2188.756445         41              среднее             1   
10         2    4171.483647         36               высшее             0   
11         0     792.701887         40              среднее             1   
12         0  365286.622650         65              среднее             1   
13         0    1846.641941         54  неоконченное высшее             2   
14         0    1844.956182         56               высшее             0   
15         1     972.364419         26              среднее             1   
16         0    1719.934226         35              среднее             1   
17         0    2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0   10038.818549         48              среднее             1   
20         1    1311.604166         36              среднее             1   
21         1     253.685166         33              среднее             1   
22         1    1766.644138         24              среднее             1   
23         0     272.981385         21               высшее             0   
24         1  338551.952911         57              среднее             1   
25         0  363548.489348         67              среднее             1   
26         0    2689.137274         41              среднее             1   
27         0     529.191635         28               высшее             0   
28         1     717.274324         26               высшее             0   
29         0  365286.622650         63              среднее             1   
30         1  335581.668515         62              среднее             1   
31         0    1682.083438         47            начальное             3   
32         0    4649.910832         34              среднее             1   
33         0    1548.637544         48              среднее             1   
34         0    4488.067031         35              среднее             1   
35         0  394021.072184         68              среднее             1   
36         0     176.216688         33              среднее             1   
37         0    6448.810860         43               высшее             0   
38         0     597.881827         25               высшее             0   
39         0     650.587796         31               высшее             0   

            family_status  family_status_id gender  income_type  debt  \
0         женат / замужем                 0      F    сотрудник     0   
1         женат / замужем                 0      F    сотрудник     0   
2         женат / замужем                 0      M    сотрудник     0   
3         женат / замужем                 0      M    сотрудник     0   
4        гражданский брак                 1      F    пенсионер     0   
5        гражданский брак                 1      M    компаньон     0   
6         женат / замужем                 0      F    компаньон     0   
7         женат / замужем                 0      M    сотрудник     0   
8        гражданский брак                 1      F    сотрудник     0   
9         женат / замужем                 0      M    сотрудник     0   
10        женат / замужем                 0

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
groupby_children = data.groupby('children')['debt'].sum()
groupby_children


children
0    1063
1     444
2     194
3      27
4       4
5       0
Name: debt, dtype: int64

**Вывод:** 

Зависимость между количеством детей и возвратом кредита в срок однозначно есть. Количество людей, у которых нет детей, но есть долги по возврату кредита, в 2 раза превышает тех, у кого есть дети и кто тоже имеет долги.

Среди людей, у которых есть дети, чаще всего долги по возврату кредита имеются у тех, у кого меньше всего детей. Чем больше человек имеет детей, тем реже у него имеются долги.

In [28]:
#создадим сводную таблицу, где в качестве столбца, по которому группируем данные будет children; в качестве столбца,
#по значениям которого происходит группировка - debt (есть/нет); значения, которые увидим в сводной 
#таблице - gender; 
#к значениям пола применим функцию подсчета количества людей - count
debt_and_kids = data.pivot_table(index=['children'], columns='debt', values='gender', aggfunc='count')
debt_and_kids


debt            0       1
children                 
0         13028.0  1063.0
1          4364.0   444.0
2          1858.0   194.0
3           303.0    27.0
4            37.0     4.0
5             9.0     NaN

In [24]:
children_and_debt_pivot = data.groupby('children')['debt'].agg(['count', 'sum', 'mean']) 
children_and_debt_pivot = children_and_debt_pivot.rename(columns={'count': 'total_borrowers', 'sum': 'debtors', 'mean': 'persentage'})
children_and_debt_pivot.sort_values(by='persentage', ascending=False)

total_borrowers  debtors  persentage
children                                      
4                      41        4    0.097561
2                    2052      194    0.094542
1                    4808      444    0.092346
3                     330       27    0.081818
0                   14091     1063    0.075438
5                       9        0    0.000000

**Вывод:**

Была построена сводная таблица, сгруппированная по **количеству детей** у клиентов, с колонками: общее количество заемщиков, количество задолжников по выплате кредита и доля должников от общего  количества.

По результатам свдной таблицы можно сделать вывод, что люди, имеющие от 1 до 4 детей имеют больше задолженностей по выплате кредитов, чем люди не имеющие детей вовсе. 

- **Должники с 1, 2 и 4 детьми составляют 9% от своих групп.**
- **Должники с 3 детьми составляют чуть меньше - 8% от своей группы.**
- **Среди клиентов без детей доля должников составляет 7%.**

____________________________________

*Также есть группа людей с 5 детьми, должников среди них нет, но количество заемщиков - всего 9 человек. Это очень маленькое количество среди общего числа заемщиков, по которому нельзя сделать какие-либо выводы, поэтому данную группу клиентов не стоит включать в анализ.*

 

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [25]:
marital_ctatus_and_debt_pivot = data.groupby('family_status')['debt'].agg(['count', 'sum', 'mean']) 
marital_ctatus_and_debt_pivot = marital_ctatus_and_debt_pivot.rename(columns={'count': 'total_borrowers', 'sum': 'debtors', 'mean': 'persentage'})
marital_ctatus_and_debt_pivot.sort_values(by='persentage', ascending=False)

total_borrowers  debtors  persentage
family_status                                              
Не женат / не замужем             2796      273    0.097639
гражданский брак                  4134      385    0.093130
женат / замужем                  12261      927    0.075606
в разводе                         1189       84    0.070648
вдовец / вдова                     951       63    0.066246

**Вывод:** 


Была построена сводная таблица, сгруппированная по **семейному положению** клиентов, с колонками: общее количество заемщиков, количество задолжников по выплате кредита и доля должников от общего  количества.

На основании построенной сводной таблицы можно сделать следующие выводы:

- **Группы клиентов, не состоящих в браке и состоящих в гражданском браке, имеют самую высокую долю должников среди своих групп - 9% в каждой.**
- **Доля должников в группе клиентов, состоящих в браке, и в группе клиентов, находящихся в разводе, значительно меньше - по 7% в каждой группе.**
- **Среди клиентов с семейным положением вдова/вдовец самая малая доля должников по выплате кредита - 6%**.


#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [29]:
income_and_debt_pivot = data.groupby('total_income_category')['debt'].agg(['count', 'sum', 'mean']) 
income_and_debt_pivot = income_and_debt_pivot.rename(columns={'count': 'total_borrowers', 'sum': 'debtors', 'mean': 'persentage'})
income_and_debt_pivot.sort_values(by='persentage', ascending=False)

total_borrowers  debtors  persentage
total_income_category                                      
E                                   22        2    0.090909
C                                15921     1353    0.084982
A                                   25        2    0.080000
B                                 5014      354    0.070602
D                                  349       21    0.060172

**Вывод:** 

Была построена сводная таблица, сгруппированная по **уровню дохода** клиентов, с колонками: общее количество заемщиков, количество задолжников по выплате кредита и доля должников от общего  количества.

Легенда к уровням дохода у клиентов:
- 0–30000 руб. — 'E';
- 30001–50000 руб. — 'D';
- 50001–200000 руб. — 'C';
- 200001–1000000 руб. — 'B';
- 1000001 руб. и выше — 'A'.

На основании сводной таблица можем сделать следющие выводы:

- **Доля должников среди клиентов с заработком от 50001 до 200000 руб. составляет 8%. При этом в данной группе самое большое число заемщиков от общего числа.**
- **В группе клиентов с заработком от 200001 до 1000000 руб. доля должиков составляет 7%**.
- **Самая низкая доля должников в группе клиентов с уровнем дохода от 30001 до 50000 руб. - всего 6%**.

*В таблице также фигурируют группы клиентов с самым низким заработком (от 0 до 30 000 руб.) и с заработком более 1000000 руб., но количество заемщико всреди обеих групп очень маленькое, в связи с чем мы не можем делать корректные выводы по данным группам.*


#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [27]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
loan_purpose_and_debt_pivot = data.groupby('purpose_category')['debt'].agg(['count', 'sum', 'mean']) 
loan_purpose_and_debt_pivot = loan_purpose_and_debt_pivot.rename(columns={'count': 'total_borrowers', 'sum': 'debtors', 'mean': 'persentage'})
loan_purpose_and_debt_pivot.sort_values(by='persentage', ascending=False)

total_borrowers  debtors  persentage
purpose_category                                              
операции с автомобилем               4279      400    0.093480
получение образования                3988      369    0.092528
проведение свадьбы                   2313      183    0.079118
операции с недвижимостью            10751      780    0.072551

**Вывод:** 

Была построена сводная таблица, сгруппированная по **целям кредита**, с колонками: общее количество заемщиков, количество задолжников по выплате кредита и доля должников от общего  количества.

На основании данной таблица можем сделать следующие выводы:

- **Самый большой процент должников среди тех, кто берут кредиты на операции с автомобилем и на получение образования. В каждой из групп по 9% клиентов с задолженностью.**
- **Те, кто берут кредиты на проведение свадьбы, имеют меньший процент должников в своей группе - 7,9%.**
- **На операции с недвижимостью клиенты чаще всего берут кредит, но доля должников среди них меньше всего - 7,2%.**

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 

Так как пропуски встретились только в двух столбцах: `days_employed` и `total_income` и количество пропущенных значений в обоих столбцах совпадало, то можем предположить, что **пропуски в обоих столбцах были в одних и тех же строчках**. Полагаю, что те, кто не заполнил общее количество дней трудового стажа, также не заполнил информацию по  своему ежемесячному доходу. 

Вероятные причины незаполненных данных в этих столбцах:

- самозанятость/фриланс
- человек не работает, а находится на содержании у мужа/жены/родителей
- отсутствие/утеря трудовой книжки

Помимо человеческого фактора, можем допустить, что возможной причиной пропусков в данных мог стать сбой в работе системы. Возможно, те люди, кто вносил данные, делал это в конкретное время, когда произошел сбой, из-за чего по каждому человеку не "подтянулсись" данные о количестве дней трудового стажа и ежемесячному дохожу. 

Полагаю, что данную информацию стоит донесит до разработчиков, чтобы они проверили корректность работы системы.

Также, когда мы выводили медианные значения количества дней трудового стажа по типу занятости, то нам встретились аномально большие значения у двух типов - безработные и пенсионеры. Вероятно, может быть какая-то взаимосвязь между пропусками и в 2 столбцах и этими двумя типами занятости.


#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 

Так как среднее значение вычиляется из суммы всех значений разделить на количество этих значений, то результат такого исчисления далеко не всегда можно считать корректным, в отличии медианы. Из-за того, что большая часть значений, которые суммируются, могут быть сильно ниже или сильно выше всех остальных значений, то среднее вычсление будет, соответственно, также либо заниженным или завышенным и не будет отражать реальное среднее значение.

Например, в столбце `days_employed` было бы некорректо считать среднее значение, если у какой-либо группы преобладает очень большое количество дней стажа и у нескольких десятков совсем небольшое. Аналогично с ежемесячной заработной платой.

Медиана же берёт *золотую середину* из выборки количественных значений, что дает данные, приближенные к реальности. 


### Шаг 4: общий вывод.

***В ходе анализа данных датафрема выявились следующие закономерности:***

**Благонадежным** по выплате кредита в срок являются те, кто: 
- имеет более от 3 и более детей, 
- находится в разводе или является вдовой/вдовцом, 
- имеет заработок от 0 до 30 000 руб. или более 1 000 000 руб. в месяц,
- берут кредит с целью отпраздновать свадьбу.

**Неблагонадежным** по выплате кредита в срок являются люди:
- без детей,
- в браке,
- с ежемесячным заработком от 50 001 до 200 000 руб.,
- берут кредит на недвижимость.
        

**Обзор и предобработка данных**

Для проведения исследования была взята таблица с данными о клиентах и их платежеспособности. Таблица состоит из следующих столбцов:
- `children` — количество детей в семье
- `days_employed` — общий трудовой стаж в днях
- `dob_years` — возраст клиента в годах
- `education` — уровень образования клиента
- `education_id` — идентификатор уровня образования
- `family_status` — семейное положение
- `family_status_id` — идентификатор семейного положения
- `gender` — пол клиента
- `income_type` — тип занятости
- `debt` — имел ли задолженность по возврату кредитов
- `total_income` — ежемесячный доход
- `purpose` — цель получения кредита

Почти все данные были полные - без пропусков, кроме двух столбцов: `days_employed` и `total_income`. В каждом из этих двух столбцов было пропущено по 2174 значения, при чем в одних и тех же строках. Пропущенные количественные значения в обоих столбцах были заменены медианными значениями для точности результатов исследования. 
Но на пропуски в обоих столбцах стоит обратить внимание разработчикам и проверить систему, откуда поступают заполненные данные, на корректность работы.

В данных также встретились некоторые аномальные значения:
1. В столбце `days_employed` были исправлены аномальные значения со знаком минус на положительные.
2. В столбца `children` встретились аномальные значения *-1* и *20*. Они были удалены из таблицы и не учитывались при проведении исследования.
3. В столбце `days_employed` при выводе медианных значений, сгруппировав данные по типу занятости `income_type`, у двух типов (безработные и пенсионеры) получились аномально большие медианные значения. Так как исправить их не представляется возможным, то было принято решение оставить их, как есть.

Была проведена обработка дубликатов: значения в столбце `education` преведены к нижнему регистру, а также найдена и удалена 71 строка дубликатов.

**Выводы по результатам исследования:**

В ходе исследования были выявлены следующие факторы, влияющие на задолженности в выплате кредита:

**1. Наличие/отсутствие детей.**
Наличие детей влияет на выплату кредита в срок. Клиенты, имеющие детей (от 1 до 4) также имеют больше задолженностей по выплате кредитов, чем люди не имеющие детей. Доля должников с 1-4 детьми состявляет 8-9%, тогда как доля должников среди клиентов без детей - 7%.

В анализе не была учтена группа клиентов, имеющих 5 детей из-за недостаточного количества данных о данной группе.

**2. Cемейное положение.**
Семейное положение также влияет на наличие задолженностей по кредиту. Среди клиенты, которые не состоят в браке или состоят в гражданском браке, самый высокий процент должников - 9%.
Среди клиентов, состоящих в браке, находящихся в разводе или овдовевших, меньший процент должников - 6-7%.

**3. Уровень дохода.**
Уровень дохода влияет на наличие задолженности по выплате кредита у клиентов.
Самая высокая доля должников выявлена среди тех клиентов, чей уровень дохода составляет от 50001 до 200000 руб.

Среди тех клиентов, кто в месяц зарабатывет от 30001 до 50000 руб. и от 200001 до 1000000 руб.  доля клиентов с задолженностью - 6-7%.

В анализе не учитывались группы с заработком от 0 до 30 000 руб. и более 1000000 руб. ввиду недостаточного количества данных по этим группам.


**4. Цель, с которой берется кредит.**
Цель, с которой берется кредит, также вляет на его выплату в срок.
Больше всего должников среди клиентов, которые берут кредит на операции с автомобилем и на получение образования - 9%.

Всего 7% процентов должников среди тех, кто берет кредит на проведение свадьбы и на операции с недвижимостью. 
Но среди этих двух категорий стоит учесть важный фактор: почти половина от общего числа клиентов берут кредит на операции с недвижимостью, тогда как на проведение свадьбы кредит берет лишь примерно 1/10 часть клиентов от общего числа. 


Таким образом на основании проведенного исследования можем составить портреты благонадежного и неблагонадежного заемщиков.

***Портрет благонадежного заемщика***: без детей, вдовец/вдова, с заработком от 30 000 до 50 000 руб.. Цель кредита: на операции с недвижимостью.

***Портрет неблагонадежного заемщика:*** имеет детей, не состоит в браке или находится в гражданском браке, с уровнем дохода от 50000 до 200000 руб. Цель кредита: на получение образования или на операции с автомобилем.


**Дополнительная рекомендация:** как ранее писала, в данных есть пропуски в колонках `days_employed` и `total_income` в одних и тех же строках. Рекомендую обратить внимание разработчиков на это и попросить их проверить корректность работы системы, где заполняются данные. 

